stacks/SchedulerStack.ts

```diff
@@ -0,0 +1,82 @@
+import { Rule } from "aws-cdk-lib/aws-events";
+import { LambdaFunction } from "aws-cdk-lib/aws-events-targets";
+import {
+  PolicyDocument,
+  PolicyStatement,
+  Role,
+  ServicePrincipal,
+} from "aws-cdk-lib/aws-iam";
+import { type StackContext, use } from "sst/constructs";
+import { Function } from "sst/constructs";
+
+import { scheduleEvent } from "@/types/schedule";
+
+import { ConfigStack } from "./ConfigStack";
+import { EventBusStack } from "./EventBusStack";
+import { StorageStack } from "./StorageStack";
+
+export function SchedulerStack({ stack }: StackContext) {
+  const { eventBus } = use(EventBusStack);
+  const secrets = use(ConfigStack);
+  const { db } = use(StorageStack);
+  const bind = [...secrets, ...(db ? [db] : [])];
+
+  const schedulerFunctionInvokeRole = new Role(
+    stack,
+    `schedulerFunctionInvokeRole-${stack.stage}`,
+    {
+      assumedBy: new ServicePrincipal("scheduler.amazonaws.com"),
+      inlinePolicies: {
+        lambdaExecute: new PolicyDocument({
+          statements: [
+            new PolicyStatement({
+              actions: ["lambda:InvokeFunction"],
+              resources: ["*"],
+            }),
+          ],
+        }),
+      },
+      roleName: `schedulerFunctionInvokeRole-${stack.stage}`,
+    },
+  );
+
+  const scheduleHandlerLambda = new Function(stack, "ScheduleHandlerLambda", {
+    handler: "packages/functions/schedule/scheduleHandlerLambda.handler",
+    bind,
+    role: new Role(stack, "PutEventsRole", {
```

Comment:
What does this do that `permissions: [eventBus]` doesn't?

Reply:
I've removed it. It is not needed.